tasks/kernel_matrix_testing/stacks.py

```diff
@@ -230,20 +192,16 @@ def ec2_instance_ids(ctx, ip_list):
 
 
 def destroy_ec2_instances(ctx, stack):
-    stack_output = os.path.join(get_kmt_os().stacks_dir, stack, "stack.output")
-    if not os.path.exists(stack_output):
-        warn(f"[-] File {stack_output} not found")
+    stack_outputs = os.path.join(get_kmt_os().stacks_dir, stack, "stack.output")
+    if not os.path.exists(stack_outputs):
         return
 
-    with open(stack_output, 'r') as f:
-        output = f.read().split('\n')
-
+    infra = build_infrastructure(stack, remote_ssh_key="")
     ips = list()
-    for o in output:
-        if not is_ec2_ip_entry(o):
-            continue
-
-        ips.append(o.split(' ')[1])
+    for arch in infra:
+        instance = infra[arch]
```

Comment:
nit: Why not `for arch, instance in infra.items()`?

Reply:
Done!